staticd/static_vrf.c

```diff
@@ -51,6 +55,37 @@ static void zebra_stable_node_cleanup(struct route_table *table,
 			static_path_list_del(&si->path_list, pn);
 			XFREE(MTYPE_STATIC_PATH, pn);
 		}
+
+		/* clean up for dst table */
+		src_table = srcdest_srcnode_table(node);
+		if (src_table) {
+			/* This means the route_node is part of the top
+			 * hierarchy and refers to a destination prefix.
+			 */
+			for (src_node = route_top(src_table); src_node;
+			     src_node = route_next(src_node)) {
+				src_si = src_node->info;
+
+				frr_each_safe (static_path_list,
+					       &src_si->path_list, src_pn) {
+					frr_each_safe (static_nexthop_list,
+						       &src_pn->nexthop_list,
+						       nh) {
+						static_nexthop_list_del(
+							&src_pn->nexthop_list,
+							nh);
+						XFREE(MTYPE_STATIC_NEXTHOP, nh);
+					}
+					static_path_list_del(&src_si->path_list,
+							     src_pn);
+					XFREE(MTYPE_STATIC_PATH, src_pn);
+				}
+
+				XFREE(MTYPE_STATIC_ROUTE, src_node->info);
```

Comment:
Don't we need another route_node_unlock(src_node) after this free?

Reply:
Yes Donald, It's not required.
Here is the flow, library takes care for resting the lock count.
route_table
```
route_table_finish->route_table_free-> tmp_node->lock = 0; -> route_node_free -> zebra_stable_node_cleanup
                                                                                 route_node_destroy
```
srcdest_table
```
route_table_finish->route_table_free-> tmp_node->lock = 0; -> route_node_free -> zebra_stable_node_cleanup
                                                                                 srcdest_rnode_destroy -> route_table_finish -> route_table_free -> tmp_node->lock = 0; -> route_node_free -> srcdest_srcnode_destroy
```

For both the nodes, lock counter is set to zero in the flow when table is going for deletion when vrf is going for delete.
So, In this flow unlock is not required.